fs/log/log.go

```diff
@@ -117,17 +124,35 @@ func InitLogging() {
 
 	// Log file output
 	if Opt.File != "" {
-		f, err := os.OpenFile(Opt.File, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0640)
-		if err != nil {
-			log.Fatalf("Failed to open log file: %v", err)
-		}
-		_, err = f.Seek(0, io.SeekEnd)
-		if err != nil {
-			fs.Errorf(nil, "Failed to seek log file to end: %v", err)
+		var logWriter io.Writer
+		if Opt.Rotate {
+			logWriter = &lumberjack.Logger{
+				Filename:   Opt.File,
+				MaxSize:    int(Opt.RotateMaxSize) / 1024 / 1024, // megabytes
+				MaxAge:     Opt.RotateMaxAge,
+				MaxBackups: Opt.RotateMaxBackups,
+				LocalTime:  false,
+				Compress:   Opt.RotateCompress,
+			}
+			// Not ideal, but lumberjack doesn't support a way to get the underlying file
```

Comment:
How would that work for for streaming stderr since the file can change at any moment?

I did think about adding a `File()` method in the lumberjack library but couldn't think of any way that we could make use of it other the `io.Copy` that I added already.

Reply:
Does that go routine solution actually work? So if you kill rclone with QUIT does the backtrace come out in the log?

Reply:
Ah, so that was the use of the Dup calls. No, that won't work with this approach. It will need to be done someway in lumberjack, either via callback on rotation (https://github.com/natefinch/lumberjack/issues/24) or via periodically checking the file (which is private right now) object.

lumberjack doesn't seem to be actively maintained, however. but there exists a fork that already has the stderr redirection: https://github.com/svatantra/lumberjack. Should we switch to that instead?

Reply:
That package appears to have copied the code straight from rclone!

Yes that is exactly what we want, but I'm not sure about the maturity of that package?

Reply:
> That package appears to have copied the code straight from rclone!

:grinning:

> but I'm not sure about the maturity of that package?

Yeah, it's not very popular (when checking imports of it). That said, the only change from the base package appears to be the stderr redirection so it should be relatively stable: https://github.com/natefinch/lumberjack/compare/v2.0...svatantra:lumberjack:v2.0